pkg/ovs/openflow/ofctrl_bridge_test.go

```diff
@@ -160,3 +162,41 @@ func TestOFBridgePacketRcvd(t *testing.T) {
 		}
 	}
 }
+
+func TestOFMeterStats(t *testing.T) {
+	// Reset OVS metrics (Prometheus) and reinitialize them to test.
+	legacyregistry.Reset()
+	metrics.InitializeOVSMetrics()
+
+	b := NewOFBridge("test-br", GetMgmtAddress(ovsconfig.DefaultOVSRunDir, "test-br"))
+	sw := newFakeOFSwitch(b)
+	b.SetOFSwitch(sw)
+
+	mpMeterStatsReply := openflow15.NewMpReply(openflow15.MultipartType_MeterStats)
+	newMeterStats := func(meterID uint32, packets uint64) *openflow15.MeterStats {
+		meterStats := openflow15.NewMeterStats(meterID)
+		bandStats := openflow15.NewMeterBandStats()
+		bandStats.PacketBandCount = packets
+		meterStats.AddBandStats(*bandStats)
+		return meterStats
+	}
+	meterStats1 := newMeterStats(1, 100)
+	meterStats2 := newMeterStats(2, 0)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats1)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats2)
+	packetCounts := make(map[int]int64)
+	checkMeterStatsReply := func(meterId int, packetCount int64) {
+		packetCounts[meterId] = packetCount
+	}
```

Comment:
This needs mutex or you can use:
```
var receivedCount1 int64
checkMeterStatsReply := func(meterID int, packetCount int64) {
        assert.Equal(t, 1, meterID)
        atomic.StoreInt64(&receivedCount1, packetCount)
}

...

assert.Eventually(t, func() bool {
         return int64(100) == atomic.LoadInt64(&receivedCount1)
}, ...
``` 
Note it's not expected to receive other meterID 2 given your code.

Reply:
Thanks! I would like to check both meters and have added the mutex.